services/wireguard/connection.go

```diff
@@ -19,27 +19,47 @@ package wireguard
 
 import (
 	"sync"
-	"time"
 
+	log "github.com/cihub/seelog"
 	"github.com/mysteriumnetwork/node/core/connection"
+	"github.com/mysteriumnetwork/node/services/wireguard/network"
 )
 
 // Connection which does no real tunneling
 type Connection struct {
-	isRunning      bool
-	noopConnection sync.WaitGroup
-	stateChannel   connection.StateChannel
+	isRunning    bool
```

Comment:
Do we really need those "isRunning" Our connection is a throw away object, and to actually make it really safe we need more than that.

Reply:
Removed.